### PR TITLE
Feat/new identities

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

**Comment by jeremad** [comment]
Please update the README as well

**Comment by jeremad** [comment]
Please rebase and update the go.mod to v3

**Review by theodelrieu — APPROVED** [review]

File: identity/generate.go [hunk 1]
```diff
@@ -58,10 +59,24 @@ func GetPublicIdentity(b64Identity string) (*string, error) {
...
-	var identity interface{}
+	identity := orderedmap.New()
 	err := b64json.Decode(b64Identity, &identity)
 	if err != nil {
 		return nil, err
 	}
+
+	_, isPrivate := identity.Get("private_encryption_key")
+	target, found := identity.Get("target")
+	if found && target == "email" && !isPrivate {
```
Comment (blastrock) [inline identity/generate.go:89]: I think you can return an error `if !found`

**Review by blastrock — APPROVED** [review]

File: identity/generate.go (hunk 1 above)
Comment (tux3) [reply identity/generate.go]: Good point, done.

**Pull request merged**